python/datafusion/dataframe.py

```diff
@@ -163,7 +163,20 @@ def with_column(self, name: str, expr: Expr) -> DataFrame:
     def with_columns(
         self, *exprs: Expr | Iterable[Expr], **named_exprs: Expr
     ) -> DataFrame:
-        """Add an additional column to the DataFrame.
+        """Add columns to the DataFrame.
+
+        By passing expressions, iteratables of expressions, or named expressions. To
+        pass named expressions use the form name=Expr.
+
+        Example usage:
+
+            The following will add 4 columns labeled a, b, c, and d.
+
+            df = df.with_columns(
+                lit(0).alias('a'),
+                [lit(1).alias('b'), lit(2).alias('c')],
+                d=lit(3)
+                )
```

Comment:
Do you know if this renders right in the online doc? If not, I can pull your branch and test it out

Reply:
@timsaucer I am not sure, is it possible to locally build the docs?

Reply:
Yes, there is a build script in the `docs` folder and then you can open the generated html files locally.